### ISSUE TITLE
Pytest is failing

### ISSUE THREAD

**Issue by ccmagruder** [charliethebrit/Tensor.py#1]
I have attached the diagnostic output when running pytest in Python 2.7. Should I be using a different version of python instead?

[pytest.txt](https://github.com/charliethebrit/Tensor.py/files/2176580/pytest.txt)

**Comment by charlescolley** [comment]
try the most recent version of 2. 

I need to look into the first and third error when I get home from work, though I think the third error is being caused that version not having a particular method. I'm also pretty sure that the first error is being caused by a number being off by something O(machine_epsilon), so I don't think these failures will block the code base usage. 

The second error is a purposeful error (note the format type is 'apple'), which is not being caught properly by pytest. 


Will try to have an update by 8-9pm tonight at the latest

**Comment by charlescolley** [comment]
Ok, I've fixed a couple of the errors, there was an issue of passing by reference instead of value. I've pushed up the changes. There's still one issue that's failing, but in order to debug that one, I need a little more time to fix it. It's related to the setting routine, which is more complicated than the other test cases. I'll comment here once I fix it. Hopefully will be able to finish the fixes this weekend.
